README.md

```diff
@@ -1,66 +1,120 @@
-  One day I might even port over at least parts of the server to it (the game API comes to mind), but not right now.
+  Unfortunately, nearly everything around it (including the runtime itself) is in beta.
 
-Python isn't the fastest thing around, but both it and its ecosystem are very stable and mature. SQLAlchemy + Alembic alone makes an incredibly robust database abstraction layer, that nothing I've seen in NodeJS can match. (Do feel free to prove me wrong though.)
+Python isn't the fastest thing around, but both it and its ecosystem are very stable and mature. SQLAlchemy + Alembic alone makes an incredibly robust database abstraction layer, that nothing I've seen in NodeJS can match. 
 
+### Q: Why PostgreSQL?
+* ***MySQL*** is [notoriously awful](http://grimoire.ca/mysql/choose-something-else), with the most important piece of incompetent nonsense being the lack of Transactional DDL… which means a failed migration (for whatever reason) may leave the database in an inconsistent state.
+* ***SQLite*** is a good second choice, but it sometimes behaves oddly, and lacks some more advanced features. It'll also slow down with a lot of data, not that a development server is really in any danger of hitting that point.
+* ***lol MS SQL and Oracle***, I hope you like enterprise nonsense.
 
 Development Setup
 ---
 
-Probably only works on Linux. Screw Ansible/Vagrant, too hard for me.
-
-1.  **Set up the environment.**  
-
-        pip install -r requirements.txt
-        sudo -u postgre createuser -s $USER
-        sudo -u postgre createdb kcsrv
+Ingredients:
+---  
+* Python 3
+* ```virtualenv```
+* PostgreSQL (+ development headers)
+
+Cooking Instructions:
+---  
+1.  **Set up the environment.**
+
+    	git clone http://github.com/kancolletool/kcsrv.git kcsrv
+    	cd kcsrv
+    	virtualenv -p /usr/bin/python3 .
+    	source ./bin/activate
+        
+    Now, enter the project's Virtual Environment. This contraption allows you to install packages locally, without conflicting with system packages. It's activated per-terminal, and must be activated before interacting with the project in any way:
+  
+        # Not a typo; `.` is an alias for `source`  
+        . bin/activate
+        
+    If you wish to exit it again, simply run:
+
+        deactivate
+        
+1. **Install dependencies.**  
+    Inside the virtualenv, simply run:
+
+    	pip install -r requirements.txt
+
+    If you ever get issues with things not being found, try re-running this and seeing what happens.
+    
+1. **Set up the database***  
+    First, make sure you have a database user for the account you plan on running the server as. Postgres can automatically authorize as the user a process is running as, which means there's no need to put passwords in configuration files.
 
+        sudo -u postgres createuser --superuser $USER
+        
+    Then, create a database for kcsrv to use:
+        sudo -u postgres createdb kcsrv  
```

Comment:
You don't need the `sudo -u postgres` part, since you just made yourself a superuser. If you really want to use it, you need to pass in your user as the database owner.